src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -9,6 +9,8 @@
 from types import SimpleNamespace
 from typing import Any, Dict, List, Optional, Tuple, TypeVar, Union
 
+from azure.mgmt.containerservice.models import KubernetesSupportPlan
```

Comment:
If you choose to update the version dependency on azure-cli to `2.51.0` (in which the change in official azure-cli is merged) (see file `azext_metadata.json`), there is no need to update the decorator here. But at the same time this will force customers to upgrade azure-cli.

Reply:
Thank you Fuming. Is there a preferred or recommended approach? or it is up to the feature owner to decide?

Reply:
I think waiting 2-3 months after azure-cli is released before updating the version dependencies is a good choice. So if you want to update to 2.51.0, better wait until October. Or you can do the code change like now.

Reply:
We would like to do the code change now, so I will keep the changes here. Thank you for the clarification!